app/src/main/java/com/alphawallet/app/ui/widget/holder/WalletSummaryHolder.java

```diff
@@ -211,15 +248,19 @@ private Wallet fetchWallet(Wallet w)
         RealmWalletData realmWallet = realm.where(RealmWalletData.class)
                 .equalTo("address", w.address)
                 .findFirst();
-
         if (realmWallet != null)
         {
             w.balance = realmWallet.getBalance();
             w.ENSname = realmWallet.getENSName();
             w.name = realmWallet.getName();
             w.ENSAvatar = realmWallet.getENSAvatar();
-        }
 
```

Comment:
If the TestNetHorizontalAdapter is converted into a widget, deriving a LinearLayout there's no need to pass the context in. See eg SettingItemsView. Keep the TestNetHorizontalAdapter class but move it into a widget, which we can simply pass a Token[] array to and it displays the tokens. Then the context pass can be removed as it has its own context.